File: cpp/2399. 检查相同字母间的距离.cpp
class Solution {
private:
    int mp[26];
public:
    bool checkDistances(string s, vector<int>& distance) {
        for(int i=0;i<26;i++){
            mp[i]=-1;
        }
        for(int i=0;i<s.length();i++){
            int t=s[i]-'a';
            if (mp[t]!=-1) mp[t]=i-mp[t]-1;
            else mp[t]=i;
        }
        for(int i=0;i<26;i++){
            if(mp[i]==distance[i]) continue;
            else if(mp[i]!=-1) return false;
        }
        return true;
    }
};

File: 刷题/golang/2399. 检查相同字母间的距离.go
func checkDistances(s string, distance []int) bool {
	firstInd := make([]int, 26)
	for i := 0; i < len(s); i++ {
		idx := s[i] - 'a'
		if firstInd[idx] != 0 && i-firstInd[idx] != distance[idx] {
			return false
		}
		firstInd[idx] = i + 1
	}
	return true
}